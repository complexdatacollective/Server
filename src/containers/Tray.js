import React, { Component } from 'react';
import { ipcRenderer } from 'electron';
import TrayMenu from '../components/TrayMenu';
import TrayMenuItem from '../components/TrayMenuItem';

class Tray extends Component {
  openMainWindow = () => {
    ipcRenderer.send('OPEN_MAIN_WINDOW');
  }

  quit = () => {
    ipcRenderer.send('QUIT');
  }

  render() {
    return (
      <div className="tray">
        <TrayMenu>
          <TrayMenuItem action={this.openMainWindow}>
            Export data
          </TrayMenuItem>
          <TrayMenuItem action={this.quit}>
            Quit
          </TrayMenuItem>
        </TrayMenu>
      </div>
    );
  }
}

export default Tray;
