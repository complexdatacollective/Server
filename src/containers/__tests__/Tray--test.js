/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import Tray from '../Tray';

describe('<Tray />', () => {
  it('It should render', () => {
    const subject = shallow(<Tray />);

    expect(subject).toMatchSnapshot();
  });

  it('It should call IPC');
});
