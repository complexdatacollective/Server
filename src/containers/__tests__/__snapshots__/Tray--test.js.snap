exports[`<Tray /> It should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="tray">
    <div
        className="tray__menu">
        <MenuItem
            action={[Function]}
            className="tray__menu-item">
            Overview
        </MenuItem>
        <MenuItem
            action={[Function]}
            className="tray__menu-item">
            Export data
        </MenuItem>
        <MenuItem
            action={[Function]}
            className="tray__menu-item">
            Quit
        </MenuItem>
    </div>
</div>,
  "nodes": Array [
    <div
      className="tray">
      <div
            className="tray__menu">
            <MenuItem
                  action={[Function]}
                  className="tray__menu-item">
                  Overview
            </MenuItem>
            <MenuItem
                  action={[Function]}
                  className="tray__menu-item">
                  Export data
            </MenuItem>
            <MenuItem
                  action={[Function]}
                  className="tray__menu-item">
                  Quit
            </MenuItem>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Tray />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Tray {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "openWindow": [Function],
        "props": Object {},
        "quit": [Function],
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="tray">
          <div
                    className="tray__menu">
                    <MenuItem
                              action={[Function]}
                              className="tray__menu-item">
                              Overview
                    </MenuItem>
                    <MenuItem
                              action={[Function]}
                              className="tray__menu-item">
                              Export data
                    </MenuItem>
                    <MenuItem
                              action={[Function]}
                              className="tray__menu-item">
                              Quit
                    </MenuItem>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="tray">
          <div
                    className="tray__menu">
                    <MenuItem
                              action={[Function]}
                              className="tray__menu-item">
                              Overview
                    </MenuItem>
                    <MenuItem
                              action={[Function]}
                              className="tray__menu-item">
                              Export data
                    </MenuItem>
                    <MenuItem
                              action={[Function]}
                              className="tray__menu-item">
                              Quit
                    </MenuItem>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Tray />,
}
`;
